BUG : `go-sqlite3` build failure - 'pread64'/'pwrite64' undeclared

## Description
The backend build fails during the compilation of `github.com/mattn/go-sqlite3`. This usually occurs in environments using `musl` libc (like Alpine) or newer `glibc` versions where the 64-bit variants of pread/pwrite are handled differently.

## Error Log
```text
sqlite3-binding.c:37644:42: error: 'pread64' undeclared here (not in a function); did you mean 'pread'?
37644 |   { "pread64",      (sqlite3_syscall_ptr)pread64,    0  },
      |                                          ^~~~~~~
sqlite3-binding.c:37662:42: error: 'pwrite64' undeclared here (not in a function); did you mean 'pwrite'?
37662 |   { "pwrite64",     (sqlite3_syscall_ptr)pwrite64,   0  },
      |                                          ^~~~~~~~
```

## Root Cause
The `go-sqlite3` version `v1.14.17` has known compatibility issues with certain system headers. Specifically, it incorrectly assumes the presence of `pread64` and `pwrite64` as distinct symbols.

## Solution
Update `github.com/mattn/go-sqlite3` to `v1.14.24` or newer. This version includes patches that correctly handle these syscalls across different libc implementations.

--- a/golang-backend/go.mod
+++ b/golang-backend/go.mod
@@ -22,7 +22,7 @@ require (
 	github.com/kelseyhightower/envconfig v1.4.0
 	github.com/lestrrat-go/jwx/v2 v2.0.11
 	github.com/lib/pq v1.10.9
-	github.com/mattn/go-sqlite3 v1.14.17
+	github.com/mattn/go-sqlite3 v1.14.24
 	github.com/pkg/errors v0.9.1
 	github.com/redis/go-redis/v9 v9.18.0
 	github.com/rubenv/sql-migrate v1.5.2

